tests/testthat/test-linter_tags.R

```diff
@@ -27,3 +27,69 @@ test_that("available_linters matches the set of linters available from lintr", {
   # ensure that all _linter objects in our namespace are also exported
   expect_setequal(all_linters, grep("_linter$", getNamespaceExports("lintr"), value = TRUE))
 })
+
+test_that("lintr help files are up to date", {
+  helper_db_dir <- system.file("help", package = "lintr")
+  skip_if_not(dir.exists(helper_db_dir))
+  skip_if_not(file.exists(file.path(helper_db_dir, "lintr.rdb")))
+  skip_if_not(file.exists(file.path(helper_db_dir, "lintr.rdx")))
+
+  help_env <- new.env(parent = topenv())
+  lazyLoad(file.path(helper_db_dir, "lintr"), help_env)
+
+  lintr_db <- available_linters()
+
+  # some of the following might seem redundant, so note: we want to establish
+  #   that entries exist in the help _if and only if_ they exist in the csv in inst/.
+  #   so e.g. checking that all linters in the DB match a line in the Rd is not enough,
+  #   e.g. if a deleted/renamed linter's entry still persists in the help page, and vice versa
+
+  expect_true(exists("linters", envir = help_env))
+  # objects in help_env are class Rd, see ?as.character.Rd (part of 'tools')
+  linter_help_text <- paste(as.character(help_env$linters), collapse = "")
+
+  # Rd markup for items looks like \item{\code{\link{...}} (tags: ...)}
+  linter_item_regex <- "[\\]item[{][\\]code[{][\\]link[{][a-zA-Z0-9._]+[}][}] [(]tags: [a-z_, ]+[)][}]"
+  expect_identical(
+    length(gregexpr(linter_item_regex, linter_help_text)[[1L]]),
+    nrow(lintr_db),
+    info = "Count of linter entries in ?linters matches inst/lintr/linters.csv row count"
+  )
+
+  tag_table <- table(unlist(lintr_db$tags))
+  all_tags <- names(tag_table)
+  tag_help_text <- character(length(all_tags))
+  names(tag_help_text) <- all_tags
+
+  for (tag in all_tags) {
+    expect_true(exists(paste0(tag, "_linters"), envir = help_env), info = paste(tag, "tag page"))
+
+    tag_help <- help_env[[paste0(tag, "_linters")]]
+    tag_help_text[[tag]] <- paste(as.character(tag_help), collapse = "")
+    # Rd markup for items looks like \item{\code{\link{...}}}
+    tag_item_regex <- "[\\]item[{][\\]code[{][\\]link[{][a-zA-Z0-9._]+[}][}][}]"
+    expect_identical(
+      length(gregexpr(tag_item_regex, tag_help_text[[tag]])[[1L]]),
+      tag_table[[tag]],
+      info = paste0("Count of linter entries in ?", tag, "_linters matches tags given in inst/lintr/linters.csv")
+    )
+  }
+
+  for (ii in seq_len(nrow(lintr_db))) {
+    # NB: matching tags requires reproducing the "platform-independent sort" we do; instead, rely on the other tests
+    #   for matching the tags
```

Comment:
WDYT about sorting the tags in `available_linters()`? That would guarantee sortedness (is that even a word?) here and allow `rd_tags()` to rely on sorted tags.

Reply:
I have kind-of been writing the tags in relevance order, which makes more sense to me than alphabetical. Don't think it makes much sense to over-orient ourselves to making the test easier to write.

capture group + sort in the test would be better.